src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -142,6 +142,30 @@ internal class AnalyzerManager
         }
 
         /// <summary>
+        /// Returns true if analyzer registered a compilation start action during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/>
+        /// which registered a compilation end action and at least one other analyzer action, that the end action depends upon.
+        /// </summary>
+        public async Task<bool> GetAnalyzerHasDependentCompilationEndAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
I recommend moving execution of compilation start actions out of AnalyzerManager rather than adding API surface to AnalyzerManager describing the behavior of compilation start actions.